src/components/atoms/Navigator/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { memo } from "react";
+
+import { useT } from "@reearth/i18n";
+import { styled } from "@reearth/theme";
+
+import Icon from "../Icon";
+
+import { useNavigator } from "./hooks";
+import { CompassAngle } from "./types";
+
+export type Props = {
+  degree: number;
+  /**
+   * Pass degree of circle as callback arguments.
+   * This event is invoked when outer circle is clicked and mouse move.
+   * If click is dismissed then this event is stopped.
+   */
+  onRotate?: (degree: number) => void;
+  /**
+   * Pass distance of movement as x,y axis(px) and degree of circle as callback arguments.
+   * This event is invoked when inner circle button is clicked and mouse move.
+   * If click is dismissed then this event is stopped.
+   */
+  onOrbit?: (angle: CompassAngle) => void;
+  /**
+   * This event is invoked when center of circle is clicked.
+   */
+  onRestoreRotate?: () => void;
+  /**
+   * Pass + or -  as callback arguments.
+   */
+  onClickHelp?: () => void;
+  onZoomIn?: () => void;
+  onZoomOut?: () => void;
+};
+
+const Navigator: React.FC<Props> = memo(function NavigatorPresenter({
+  degree,
+  onRotate,
+  onOrbit,
+  onRestoreRotate,
+  onClickHelp,
+  onZoomIn,
+  onZoomOut,
+}) {
+  const {
+    compassRef,
+    compassDegree,
+    compassFocusAngle,
+    isMovingAngle,
+    handleOnMouseDownAngle,
+    handleOnMouseDownCompass,
+  } = useNavigator({ degree, onRotate, onOrbit });
+  const t = useT();
+  return (
+    <Container>
+      <CompassContainer>
+        <Compass ref={compassRef}>
+          <>
+            <CompassIcon onMouseDown={handleOnMouseDownCompass}>
+              <Icon
+                icon="compass"
+                color="#000"
+                aria-label={t("compass")}
+                size={64}
+                style={{ transform: `rotate(${compassDegree}deg)` }}
+              />
+            </CompassIcon>
```

Comment:
Are you able to make a `const CompassIcon = styled(Icon)`...styles`;` and avoid wrapping the Icon in a div? Or does the Icon component not give you access to onMouseDown?
If the styled(Icon) is possible, it reduces our nested div count a little.. :P

Reply:
I noticed you are doing this wrapper with all the Icons in this component. If you can do styled(Icon), please update all the other icons too, which will reduce the component size by quite a bit

Reply:
There are some reason why I don't use `styled(Icon)` pattern like the following:pray:

- `<AngleIcon/>` and `<CompassIcon/>` need to set `onMouseDown` event, but `<Icon/>` does not take event.
- `<CompassFocusIcon/>` has different size to `<Icon />` size. And `<CompassFocusIcon/>` has animation that rotate around the center of compass while `<CompassFocusIcon/>` keep size. To accomplish this animation, we need to define `<CompassFocusIcon/>`.
- `<ToolIconButton/>` is button element.

Reply:
@keiya01 Thanks for being careful, but if you have the time I think it would be great to update the Icon component to accept onMouseDown/Up events. 
For the rotation, can you not just set transform in the styled icon: `styled(Icon)`HERE``? If not, maybe adding a transition prop to Icon would work?
If you find an atom component doesn't meet your needs, and adding a prop seems beneficial to the atom AND doesn't over complicate it feel free to modify them when you need!

Reply:
https://github.com/reearth/reearth-web/pull/308/commits/cd21558514e20b788c00f28fa295599f2edfb530
I fixed this review!

Reply:
Hmm, if I set `onMouseDown` event to svg element then test fail... But `Navigator` works fine in storybook.
I don't know why this behavior occurs this problem but maybe this is occurred by jest-dom.
Do you know about this behavior?

Reply:
Testing svg element is difficult so I think we can keep `div` element. Do you think about this?

Reply:
I have noticed another unexpected behaviour around Icons when passing some props to an Icon using styled components, so I am thinking of removing the svg dep we are currently using and try making the Icons component to be similar to what we are doing in another repo, that seems to be working well.
For you and this PR, if it works in storybook well, and nothing breaks if you extensively use it there, then I think it is fine!